src/datamigration/azext_datamigration/manual/custom.py

```diff
@@ -0,0 +1,245 @@
+from knack.util import CLIError
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Assessment Command Implementation.
+# -----------------------------------------------------------------------------------------------------------------
+def datamigration_assessment(connection_string=None,
+                             output_folder=None,
+                             overwrite=False,
+                             config_file_path=None):
+
+    try:
+
+        if not platform.system().__contains__('Windows'):
+            raise CLIError("This command cannot be run in non-windows environment. Please run this command in Windows environement")
+
+        defaultOutputFolder = get_default_output_folder()
+
+        # Assigning base folder path
+        baseFolder = os.path.join(defaultOutputFolder, "Downloads")
+        exePath = os.path.join(baseFolder, "SqlAssessment.Console.csproj", "SqlAssessment.exe")
+
+        # Creating base folder structure
+        if not os.path.exists(baseFolder):
+            os.makedirs(baseFolder)
+
+        testPath = os.path.exists(exePath)
+
+        # Downloading console app zip and extracting it
+        if not testPath:
+            zipSource = "https://sqlassess.blob.core.windows.net/app/SqlAssessment.zip"
+            zipDestination = os.path.join(baseFolder, "SqlAssessment.zip")
+
+            urllib.request.urlretrieve(zipSource, filename=zipDestination)
+            with ZipFile(zipDestination, 'r') as zipFile:
+                zipFile.extractall(path=baseFolder)
+
+        if connection_string is not None:
+            if output_folder is not None:
+                if overwrite is True:
+                    cmd = f'{exePath} Assess --sqlConnectionStrings "{connection_string}" --outputFolder {output_folder}'
+                else:
+                    cmd = f'{exePath} Assess --sqlConnectionStrings "{connection_string}" --outputFolder {output_folder} --overwrite False'
+                subprocess.call(cmd, shell=False)
+            else:
+                if overwrite is True:
+                    cmd = f'{exePath} Assess --sqlConnectionStrings "{connection_string}"'
+                else:
+                    cmd = f'{exePath} Assess --sqlConnectionStrings "{connection_string}" --overwrite False'
+                subprocess.call(cmd, shell=False)
+        elif config_file_path is not None:
+            test_config_file_path(config_file_path)
+            cmd = f'{exePath} --configFile "{config_file_path}"'
+            subprocess.call(cmd, shell=False)
+        else:
+            raise CLIError('No valid parameter set used. Please provide any one of the these prameters: connection_string, config_file_path')
+
+        # Printing log file path
+        logFilePath = os.path.join(defaultOutputFolder, "Logs")
+        print(f"Event and Error Logs Folder Path: {logFilePath}")
+
+    except Exception as e:
+        raise e
```

Comment:
You can rise CLI error right?

Reply:
I have left it as general because if the error is raised by any other modules like urllib or zipfile. We will also be able to catch it and raise it in that case.